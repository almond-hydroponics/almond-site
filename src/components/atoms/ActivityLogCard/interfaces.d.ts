import { ReactNode } from 'react';

export interface ActivityLogCardProps {
	log: string | ReactNode;
	date?: Date | string;
	type: string;
}
