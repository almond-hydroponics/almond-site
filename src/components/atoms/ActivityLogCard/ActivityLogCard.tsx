import dayjs from 'dayjs';
// interfaces
import { ActivityLogCardProps } from './interfaces';
import { Card, CardContent, Typography } from '@mui/material';

const ActivityLogCard = ({
	log,
	date,
	type,
}: ActivityLogCardProps): JSX.Element => {
	return (
		<Card variant="outlined">
			<CardContent>
				<Typography variant="body2" data-testid="header">
					{log}
				</Typography>
				<Typography
					variant="caption"
					style={{ fontWeight: 600 }}
					data-testid="details"
				>
					{`${dayjs(date).format('HH:mm:ss')}`}
				</Typography>
			</CardContent>
		</Card>
	);
};

export default ActivityLogCard;
