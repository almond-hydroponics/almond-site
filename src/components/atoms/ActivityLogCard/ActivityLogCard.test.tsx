// react libraries
import { render, screen } from '@testing-library/react';
import dayjs from 'dayjs';
// component
import ActivityLogCard from './ActivityLogCard';

describe('ActivityLogCard component', () => {
	const props = {
		log: 'Pump broken',
		date: '2019-10-30T08:00:42.767Z',
		type: 'info',
	};

	it('should render correctly', () => {
		const { asFragment } = render(<ActivityLogCard {...props} />);
		expect(asFragment()).toMatchSnapshot();

		const elemHeader = screen.getByTestId('header');
		expect(elemHeader.innerHTML).toBe('Pump broken');

		const elemDetails = screen.getByTestId('details');
		expect(elemDetails.innerHTML).toBe(dayjs(props.date).format('HH:mm:ss'));
	});
});
