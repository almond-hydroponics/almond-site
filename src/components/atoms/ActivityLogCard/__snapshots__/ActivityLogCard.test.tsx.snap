// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityLogCard component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 makeStyles-info-5 MuiPaper-outlined MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root makeStyles-content-3"
    >
      <p
        class="MuiTypography-root MuiTypography-body2"
        data-testid="header"
      >
        Pump broken
      </p>
      <span
        class="MuiTypography-root MuiTypography-caption"
        data-testid="details"
        style="font-weight: 600;"
      >
        11:00:42
      </span>
    </div>
  </div>
</DocumentFragment>
`;
